src/components/CodeSnippet.vue

```diff
@@ -0,0 +1,147 @@
+    return {
+      selectedTab: 0
+    };
+  },
+  computed: {
+    snippetTabs() {
+      const tabs = [
+        {label: "gemmapy", language: "python", instructions: 'Install the gemmapy package from Github and run the following code in a Python console.'}, 
+        {label: "gemma.R", language: "r", instructions: `Install the gemma.R package from BioConductor and run the following code in an R console.` },
+        {label: "curl", language: "bash", instructions: `Run the following code in a terminal (limit 100 datasets).`},
+        {label: "HTTP/1.1", language: "http", instructions: `To use with your favourite HTTP client.`}
+      ];
+
+      // Modify the content based on the searchSettings prop
+      let query = this.browsingOptions.query;
+      let filter = this.browsingOptions.filter;
+      let sort = this.browsingOptions.sort;
+
+      // Gemmapy snippet
+      let queryGemmapy = [];
+      if (query !== undefined){ queryGemmapy.push(`query = '` + query + `', `) }; 
+      if (filter !== undefined && filter.length > 0){ queryGemmapy.push(`filter = '` + filter + `', `) };
+      if (queryGemmapy.length > 0) {
+        if (sort !== undefined){ queryGemmapy.push(`sort = '` + sort + `', offset = offset, limit = '100'`) };
+        queryGemmapy.unshift(`import gemmapy\n` +
+                              `api_instance = gemmapy.GemmaPy()\n` + 
+                              `offset = 0\n` + 
+                              `all_datasets = []\n` + 
+                              `while True:\n` +
+                              `\tapi_response = api_instance.get_datasets_by_ids([],`);
+        queryGemmapy.push(`)\n` +
+                              `\tif api_response.data:\n` +
+                                `\t\tall_datasets.extend(api_response.data)\n` +
+                                `\t\toffset += 100\n` +
+                              `\telse:\n` +
+                                `\t\tbreak`);
+      } else {
+         queryGemmapy = ["No filters selected."];
+      }
+      tabs[0].content = queryGemmapy.join("").replace(/\,\s*\)/, ')');
+
+      // Gemma.R snippet
+      let queryGemmaR = [];
+      if (query !== undefined){ queryGemmaR.push(`query = '` + query + `', `) }; 
+      if (filter !== undefined && filter.length > 0){ queryGemmaR.push(`filter = '` + filter + `', `) };
+      if (queryGemmaR.length > 0) {
+        if (sort !== undefined){ queryGemmaR.push(`sort = '` + sort + `', `) };
+        queryGemmaR.unshift(`library(gemma.R)\n` +
+                            `library(dplyr)\n` + 
+                            `data <- get_datasets_by_ids(`);
+        queryGemmaR.push(`) %>% \n` +
+        `gemma.R:::get_all_pages()`);
+      } else {
+         queryGemmaR = ["No filters selected."];
+      }
+      tabs[1].content = queryGemmaR.join("").replace(/\,\s*\)/, ')');
+
+      // curl snippet
+      let encodedQuery = '';
+      if (query !== undefined) {
+        encodedQuery = 'query=' + encodeURIComponent(query);
```

Comment:
You should use [URLSearchParams](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams) API to produce the URL query.